### DISCUSSION TITLE
weird output from ollama in Librechat, but correct output from ollama in command line

### DISCUSSION THREAD

**Discussion by jowin202** [danny-avila/LibreChat#3631] [Troubleshooting]
### What happened?

Hello,

I cloned Librechat (tested v0.7.3 and v0.7.4) from git. 
But when I ask some usual things like the German "What's the meaning of life?", I get this on the following:

![image](https://github.com/user-attachments/assets/72bd0913-9d17-4ec5-a20a-4ff9082ab167)

This text is obviously random words and doesn't make sense in any way. However, when I ask the same question on command line in ollama container, then I get a proper answer. This means the ollama model should be correct, but the answer I get in librechat isn't. (text in screenshot is cut, but I fully read it and it totally makes sense in german).

![image](https://github.com/user-attachments/assets/7fa4c27b-c7bc-406c-8e47-a77a03dd7a43)

### Steps to Reproduce

I get a blank copy of Librechat (tested v0.7.3 and v0.7.4)

I changed following in the .env file
- LDAP settings for our custom server (login works properly)
- ENDPOINTS=custom (no public providers like openAI should be displayed in menu, Ollama only)

Changes in docker-compose.override.yml (syntax is ok, there were no error msgs)
- uncomment "use librechat config file"
- uncomment local build (want to change the logo later)
- uncomment the ollama part (ollama container should be installed by docker compose)

Changes in librechat.yaml (which is basically from github)
```
endpoints:
  custom:
    - name: "Ollama"
      apiKey: "ollama"
      baseURL: "http://host.docker.internal:11434/v1/"
      models:
        default: [
          "llama3"
        ]
        fetch: false # fetching list of models is not supported
      titleConvo: true
      titleModel: "current_model"
      summarize: false
      summaryModel: "current_model"
      forcePrompt: false
      modelDisplayLabel: "Ollama"
      addParams:
          "stop": [
              "<|start_header_id|>",
              "<|end_header_id|>",
              "<|eot_id|>",
              "<|reserved_special_token"
          ]
```

docker compose up -d works and all containers are running (all images removed in advance. so everything was downloaded or newly built.)

Then I go to the ollama docker container (docker exec -it ... bash) and install llama3 model by
ollama pull llama3

This is my whole setup. Then I login on web interface with my LDAP user and ask the question "Was ist der Sinn des Lebens?" (whats the meaning of life in german) and I got the screenshot from above

with ollama run llama3, I can ask questions in command line. In this case I receive a proper answer.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I recommend adding the dropParams field and dropping temperature unless you mean to experiment with this feature, along with the other model parameters


https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/custom_endpoint#dropparams

**Reply by jowin202** [reply]
Ok thank you very much :) I’ll try this @ work today 

but why does this happen? I set up a server 3 weeks ago exactly the same way and nothing like this happened. At this time I did not care about the exact GitHub commit. I just downloaded the current status. Could some recent change in some component have caused this?

**Reply by danny-avila** [reply]
@jowin202 perhaps, I need to double check this. The behavior should be that if the default values for the model parameters are not touched, that is they are left default, then they shouldn’t be sent to the API. Going to take a hard look at Ollama in general soon

**Comment by danny-avila** [comment]
Ok finally got around to this. It was indeed the model defaults being passed to Ollama. I decided to retire this functionality app-wide, as it only causes random issues like this that shouldn't happen

So in short this PR will fix it without needing the additional config: https://github.com/danny-avila/LibreChat/pull/3682
